### ISSUE TITLE
Kernel grad in acquisition function

### ISSUE THREAD

**Issue by KrisNguyen135** [sarmueller/gibo#3]
Hello,

I noticed the the kernel grad w.r.t. the input `x` has two separate implementations:
- One in the model (https://github.com/sarmueller/gibo/blob/main/src/model.py#L225-L245) where `X` is the training set.
- The other in the acquisition function (https://github.com/sarmueller/gibo/blob/main/src/acquisition_function.py#L41-L61) where `X` could be anything.

However, the first one squares the length scale on line 240 while the other doesn't on line 56. As far as I know, the length scale should indeed be squared in the kernel grad, so is this a bug in the acquisition function?

Thanks in advance,
Quan

**Comment by avrohr** [comment]
Hello Quan,

yes, that is a bug in the implementation of the acquisition function. Thank you for making us aware. We are currently discussing how to handle this situation as we want to keep the repository intact so our results are exactly reproducible. We have rerun GIBO with the fixed acquisition function and found that GIBOs performance increases slightly when we square the lengthscales correctly.

Thank you again and sorry for the very delayed replay,
Alexander von Rohr

**Comment by KrisNguyen135** [comment]
Hey Alexander, thanks for the note!

I think at one point I also realized that the missing length scale $\ell$ in the acquisition function is a multiplicative factor, so part of the posterior covariance matrix for the gradient (https://github.com/sarmueller/gibo/blob/main/src/acquisition_function.py#L104) is just a scaled version of the correct matrix, and since we're taking the trace afterwards, this wouldn't change the acquisition function maximizer if we had an isotropic kernel.

I don't think this would be the case for an ARD kernel, however. But perhaps the learned kernel is usually isotropic and you won't see a big difference between the two versions.

Anyway, I thought I should mention that I really like this work, and a couple of colleagues and I found a small room for improvement in a new paper (https://arxiv.org/abs/2210.11662) 😃 

Best,
Quan
